distributed/tests/test_client.py

```diff
@@ -2371,15 +2374,15 @@ async def test_cancel_multi_client(s, a, b):
 
 
 @gen_cluster(nthreads=[("", 1)], client=True)
-async def test_cancel_before_known_to_scheduler(c, s, a):
-    with captured_logger("distributed.scheduler") as slogs:
-        f = c.submit(inc, 1)
-        await c.cancel([f])
+async def test_cancel_before_known_to_scheduler(c, s, a, caplog):
+    caplog.set_level(logging.INFO, logger="distributed.scheduler")
+    f = c.submit(inc, 1)
+    await c.cancel([f])
 
-        with pytest.raises(CancelledError):
-            await f
+    with pytest.raises(CancelledError):
+        await f
 
-        assert "Scheduler cancels key" in slogs.getvalue()
```

Comment:
I don't really care here, but this seems like a significant relaxation of the test

Reply:
absolutely. this was an accident

Reply:
I don't entirely understand why I need to wait here since I'd expect the logging module to go through the handlers synchronously.

FWIW the usage of caplog is unrelated. Same problem appears with the `captured_logger` ctx